app/uk/gov/hmrc/customs/managesubscription/services/SubscriptionCompleteBusinessService.scala

```diff
@@ -67,6 +78,30 @@ class SubscriptionCompleteBusinessService @Inject() (
       _         <- emailService.sendFailureEmail(recipient.recipientDetails)
     } yield (): Unit
 
+  private def dataStoreEmailRequest(
+    recipient: RecipientDetailsWithEori
+  )(implicit hc: HeaderCarrier): String => Future[HttpResponse] = {
+    eori =>
+      dataStoreConnector.updateDataStore(
+        DataStoreRequest(eori, recipient.recipientDetails.recipientEmailAddress, recipient.emailVerificationTimestamp)
+      )
+  }
+
+  private def retrieveEori(recipient: RecipientDetailsWithEori)(implicit hc: HeaderCarrier): Future[Option[String]] = {
+    lazy val buildQueryParams: List[(String, String)] = {
+      val generateUUIDAsString: String = UUID.randomUUID().toString.replace("-", "")
+      List("regime" -> "CDS", "acknowledgementReference" -> generateUUIDAsString)
+    }
+    // TODO Remove gaurd for CDS clients only - do for everyone instead?
+//    if (recipient.recipientDetails.enrolmentKey != "HMRC-CUS-ORG")
+//      Future.successful(None)
+//    else
+    if (recipient.eori.isDefined)
+      Future.successful(recipient.eori)
+    else
+      subscriptionDisplayConnector.callSubscriptionDisplay(("taxPayerID" -> recipient.safeId) :: buildQueryParams)
```

Comment:
Because I don't think we should do this just for CDS. I believe we should do it for everybody... The moment there is a CDS Contract in ETMP I think we should also have an email address in customs data store... This is because CDS Financials need that email to contact traders to tell them about reimbursements etc. When the trader subscribes to CDS they will go through the short journey where we do not add their email to customs-data-store and so CDS Financials will not be able to email them.

Yashar is worried about PII. keeping the email of users without relling them, and just incase they subscribe to CDS... Which brings us back to what the architects said in the first place... Having a CDS Contract in ETMP makes no sense unless you subscribe to CDS too... So creating a contract for ATA, GVMS etc, makes no sense unless those very same orgs will some day end up doing customs declarations and hence will need access to CDS Financials. 

Will discuss this with the architects... I think ECC has become a bit confused. Call me when you have time to discuss also...